File: src/main/java/com/github/zametki/model/ZametkaId.java
package com.github.zametki.model;

import com.github.mjdbc.DbMapper;
import com.github.mjdbc.Mapper;

public class ZametkaId extends AbstractId {
    public ZametkaId(int value) {
        super(value);
    }

    @Mapper
    public static final DbMapper<ZametkaId> MAPPER = r -> new ZametkaId(r.getInt(1));
}

File: src/main/java/com/github/zametki/model/ZametkaId.kt
package com.github.zametki.model

import com.github.mjdbc.DbMapper
import com.github.mjdbc.Mapper

class ZametkaId(value: Int) : AbstractId(value) {
    companion object {

        @Mapper
        @JvmField
        val MAPPER = DbMapper { ZametkaId(it.getInt(1)) }
    }
}
